omnigibson/envs/env_base.py

```diff
@@ -601,8 +614,17 @@ def step(self, action):
                 - bool: truncated, i.e. whether this episode ended due to a time limit etc.
                 - dict: info, i.e. dictionary with any useful information
         """
+        # Pre-processing before stepping simulation
         self._pre_step(action)
+
+        # Step simulation
         og.sim.step()
+
+        # Render any additional times requested
+        for _ in range(n_render_iterations - 1):
+            og.sim.render()
```

Comment:
Same question as above. Doesn't step already take care of the additional rendering?

Reply:
No, it's separate issues:

- We want multiple rendering steps per action step merely for the sake of human comfort. As you mentioned, if headless mode is enabled, there's no reason for rendering_dt =/= sim_step_dt
- We want our observations to be accurate wrt to the most up-to-date representation of the world. This requires multiple renders to occur after **all** physics steps have been taken, but **before** observations are aggregated.

Reply:
resolved in person